AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -93,77 +97,79 @@ - (void)enqueueItem:(id<MSLog>)item flags:(MSFlags)flags {
 
   // Internal ID to keep track of logs between modules.
   NSString *internalLogId = MS_UUID_STRING;
-
+  __weak typeof(self) weakSelf = self;
   @autoreleasepool {
 
     // Additional preparations for the log. Used to specify the session id and distribution group id.
     [self enumerateDelegatesForSelector:@selector(channel:prepareLog:)
                               withBlock:^(id<MSChannelDelegate> delegate) {
-                                [delegate channel:self prepareLog:item];
+                                [delegate channel:weakSelf prepareLog:item];
                               }];
 
     // Notify delegate about enqueuing as fast as possible on the current thread.
     [self enumerateDelegatesForSelector:@selector(channel:didPrepareLog:internalId:flags:)
                               withBlock:^(id<MSChannelDelegate> delegate) {
-                                [delegate channel:self didPrepareLog:item internalId:internalLogId flags:flags];
+                                [delegate channel:weakSelf didPrepareLog:item internalId:internalLogId flags:flags];
                               }];
   }
 
   // Return fast in case our item is empty or we are discarding logs right now.
   dispatch_async(self.logsDispatchQueue, ^{
     // Use separate autorelease pool for enqueuing logs.
     @autoreleasepool {
+      typeof(self) strongSelf = weakSelf;
 
       // Check if the log should be filtered out. If so, don't enqueue it.
       __block BOOL shouldFilter = NO;
-      [self enumerateDelegatesForSelector:@selector(channelUnit:shouldFilterLog:)
-                                withBlock:^(id<MSChannelDelegate> delegate) {
-                                  shouldFilter = shouldFilter || [delegate channelUnit:self shouldFilterLog:item];
-                                }];
+      [strongSelf enumerateDelegatesForSelector:@selector(channelUnit:shouldFilterLog:)
+                                      withBlock:^(id<MSChannelDelegate> delegate) {
+                                        shouldFilter = shouldFilter || [delegate channelUnit:weakSelf shouldFilterLog:item];
```

Comment:
This seems unnecessary change based on current implementation but we don't need to have a strong reference here as well.

Reply:
When a strongReference is defined in the bloc it's intuitive to always use it instead of the weak reference. Now we have to mix and match and it introduces some complexity so I was wondering if it was worth it.

Reply:
Revert unnecessary changes in this file - #2001